server/app/views/BaseHtmlView.java

```diff
@@ -241,17 +250,26 @@ private LiTag renderPreviousPageButton(int page, Function<Integer, Call> linkFor
                         .withClasses("usa-icon", "h-4", "w-4")
                         .attr("role", "img")
                         .attr("aria-hidden", "true"),
-                    span("Previous").withClass("usa-pagination__link-text")));
+                    span(optionalMessages.isPresent()
+                            ? optionalMessages
+                                .get()
+                                .at(MessageKey.BUTTON_PREVIOUS_SCREEN.getKeyName())
+                            : "Previous")
```

Comment:
suggestion (non-blocking): Is there a way to always pass in messages and have it fall back to the english translation?

Reply:
The issue here is that `ProgramApplicationListView.java` also calls `renderPagination` but, since it's admin-facing, doesn't have access to messages, so I need to provide a way to call that method without messages:  [ProgramApplicationListView.java](https://github.com/civiform/civiform/blob/385110b91a1ed1e59000927fbb7dcfcf7fc8dd06/server/app/views/admin/programs/ProgramApplicationListView.java#L126).  Does that make sense, or is there another way you were thinking of?

Reply:
Yep that makes sense. I was hoping there was a way to access messages from the other invocation to avoid all the ternaries, but sounds like not at this time! Thanks for explaining, I'll resolve this.